Modules/_hashopenssl.c

```diff
@@ -749,36 +869,36 @@ static PyType_Spec EVPXOFtype_spec = {
 #endif
 
 static PyObject *
-EVPnew(PyObject *module, const EVP_MD *digest,
+EVPnew(PyObject *module, PY_EVP_MD *digest,
```

Comment:
As this always frees digest on every return path, we should mention that in the name, `_Py_EVPnew_and_free_digest` would make it a lot more clear who owns digest when reading code.